### PR TITLE
Bug 1712086 validate messages from pontoon addon

### PR DESCRIPTION
**Pull request by MikkCZ**
I just typed this code blindly, didn't have time to run compilation or tests locally yet.

### CODE REVIEW

**Review by Pike — CHANGES_REQUESTED** [review]
We should wrap the `JSON.parse` in a try-catch block to allow for non-JSON string messages to be posted.

**Comment by MikkCZ** [comment]
After this PR (and reading several pages of the Flow docs) I know I will stay with TypeScript. :)

**Comment by MikkCZ** [comment]
@Pike please re-review, I updated the branch from master.

File: frontend/src/modules/addonpromotion/components/AddonPromotion.tsx [hunk 1]
```diff
@@ -39,13 +46,33 @@ export class AddonPromotionBase extends React.Component<InternalProps, State> {
...
+                try {
+                    data = JSON.parse(
+                        event.data as string,
+                    ) as PontoonAddonInfoMessage;
+                } catch (_) {
+                    return;
+                }
+                break;
+        }
+        if (data && data._type === 'PontoonAddonInfo' && data.value) {
```
Comment (Pike) [inline frontend/src/modules/addonpromotion/components/AddonPromotion.tsx:73]: Can you use optional chaining here and below?

```
if (data?._type === 'PontoonAddonInfo' && data.value?.installed === true)
```

I doubt we can do so in our js code base yet, but in the TS one, this should be transpiled away.

Comment (MikkCZ) [reply frontend/src/modules/addonpromotion/components/AddonPromotion.tsx]: It's done. I touched another place in the code as well, that deserved to be polished after migration to TypeScript.

**Review by Pike — APPROVED** [review]

File: frontend/src/modules/addonpromotion/components/AddonPromotion.tsx (hunk 1 above)
Comment (Pike) [reply frontend/src/modules/addonpromotion/components/AddonPromotion.tsx]: Sweet, thanks for removing the ts-expect-error.

**Review by mathjazz — APPROVED** [review]
\o/

**Pull request merged**